js/app.js

```diff
@@ -1,6 +1,164 @@
+    return;
+  };
+
+  $scope.moveLeft = function(idx) {
+    temp = $scope.vegHolder.splice(idx, 1).toString();;
+    $scope.foodArray.push(temp);
+    $scope.winCheck();
+    return;
+  };
+
+  $scope.moveUpMid = function(idx) {
+    if($scope.foodArray[idx] !== $scope.foodArray[0]) {
+      temp = $scope.foodArray[idx];
+      temp2 = $scope.foodArray[idx - 1];
+      $scope.foodArray[idx] = temp2;
+      $scope.foodArray[idx - 1] = temp;
+    }
+    $scope.winCheck();
+    return $scope.foodArray;
+  };
+
+  $scope.moveUpRight = function(idx) {
+    if($scope.vegHolder[idx] !== $scope.vegHolder[0]) {
+      temp = $scope.vegHolder[idx];
+      temp2 = $scope.vegHolder[idx - 1];
+      $scope.vegHolder[idx] = temp2;
+      $scope.vegHolder[idx - 1] = temp;
+    }
+    $scope.winCheck();
+    return $scope.veghHolder;
+  };
+
+  $scope.moveUpLeft = function(idx) {
+    if($scope.fruitHolder[idx] !== $scope.fruitHolder[0]) {
+      temp = $scope.fruitHolder[idx];
+      temp2 = $scope.fruitHolder[idx - 1];
+      $scope.fruitHolder[idx] = temp2;
+      $scope.fruitHolder[idx - 1] = temp;
+    }
+    $scope.winCheck();
+    return $scope.fruitHolder;
+  };
+
+  $scope.moveDownMid = function(idx) {
+    if($scope.foodArray[idx] !== $scope.foodArray[$scope.foodArray.length - 1]) {
+      console.log('in')
+      temp = $scope.foodArray[idx];
+      console.log(temp)
+      console.log($scope.foodArray)
+      temp2 = $scope.foodArray[idx + 1];
+      $scope.foodArray[idx] = temp2;
+      $scope.foodArray[idx + 1] = temp;
+    }
+    $scope.winCheck();
+    return $scope.foodArray;
+  };
+
+  $scope.moveDownLeft = function(idx) {
+    if($scope.fruitHolder[idx] !== $scope.fruitHolder[$scope.fruitHolder.length - 1]) {
+      temp = $scope.fruitHolder[idx];
+      temp2 = $scope.fruitHolder[idx + 1];
+      $scope.fruitHolder[idx] = temp2;
+      $scope.fruitHolder[idx + 1] = temp;
+    }
+    $scope.winCheck();
+    return $scope.fruitHolder;
+  };
+
+  $scope.moveDownRight = function(idx) {
+    if($scope.vegHolder[idx] !== $scope.vegHolder[$scope.vegHolder.length - 1]) {
+      console.log('hi')
+      temp = $scope.vegHolder[idx];
+      temp2 = $scope.vegHolder[idx + 1];
+      $scope.vegHolder[idx] = temp2;
+      $scope.vegHolder[idx + 1] = temp;
+    }
+    $scope.winCheck();
+    return $scope.vegHolder;
+  };
+
+  $scope.winCheck = function() {
+      temp = $scope.vegHolder;
+      temp = temp.toString();
+      vegtemp = vegetables.toString();
+      if(temp === vegtemp && boolean === false) {
+        console.log(temp + ' vegholder');
+        alert('veg wins;D');
+        boolean = true;
+        return;
+      };
+      temp2 = $scope.fruitHolder;
+      temp2 = temp2.toString();
+      fruittemp = fruits.toString();
+      if(temp2 === fruittemp && anotherboolean === false) {
+        console.log(temp2 + ' fruitholder');
+        alert('fruit wins;D');
+        anotherboolean = true;
+        return;
+      };
+  };
```

Comment:
here is my win check function:
    
    $scope.winCheck = function(){
        
        for (var i = 0; i < $scope.allFruit.length; i++) {
            if ($scope.originalFruit.indexOf($scope.allFruit[i]) == -1) {
                $scope.fail = true;
                return false;
            }
        }
        for (var i = 0; i < $scope.allVegetables.length; i++) {
            if ($scope.originalVegetables.indexOf($scope.allVegetables[i]) == -1) {
                $scope.fail = true;
                return false;
            }
        }
        $scope.win = true;
        console.log($scope.win);
    }